src/sidebarApps/extensions/index.js

```diff
@@ -235,28 +240,88 @@ async function searchPlugin() {
 }
 
 async function filterPlugins() {
-	const filterOptions = {
-		[strings.top_rated]: "top_rated",
-		[strings.newly_added]: "newest",
-		[strings.most_downloaded]: "downloads",
+	const verifiedLabel = strings["verified publisher"];
+	const authorLabel = strings.author || strings.name;
+	const keywordsLabel = strings.keywords;
+
+	const filterItems = [
+		{ value: "orderBy:top_rated", text: strings.top_rated },
+		{ value: "orderBy:newest", text: strings.newly_added },
+		{ value: "orderBy:downloads", text: strings.most_downloaded },
+		{ value: "attribute:verified", text: verifiedLabel },
+		{ value: "attribute:author", text: authorLabel },
+		{ value: "attribute:keywords", text: keywordsLabel },
+	];
+
+	const filterConfig = {
+		"orderBy:top_rated": {
+			type: "orderBy",
+			value: "top_rated",
+			baseLabel: strings.top_rated,
+		},
+		"orderBy:newest": {
+			type: "orderBy",
+			value: "newest",
+			baseLabel: strings.newly_added,
+		},
+		"orderBy:downloads": {
+			type: "orderBy",
+			value: "downloads",
+			baseLabel: strings.most_downloaded,
+		},
+		"attribute:verified": {
+			type: "verified",
+			baseLabel: verifiedLabel,
+			value: true,
+		},
+		"attribute:author": { type: "author", baseLabel: authorLabel },
+		"attribute:keywords": { type: "keywords", baseLabel: keywordsLabel },
 	};
 
-	const filterName = await select("Filter", Object.keys(filterOptions));
-	if (!filterName) return;
+	const selection = await select("Filter", filterItems);
+	if (!selection) return;
 
-	$searchResult.content = "";
-	const filterParam = filterOptions[filterName];
-	currentFilter = filterParam;
-	filterCurrentPage = 1;
+	const option = filterConfig[selection];
+	if (!option) return;
+
+	const filterState = {
+		...option,
+		nextPage: 1,
+		buffer: [],
+		hasMoreSource: true,
+		displayLabel: option.baseLabel,
+	};
+
+	if (option.type === "author") {
+		const authorName = (await prompt("Enter author name", "", "text"))?.trim();
+		if (!authorName) return;
+		filterState.value = authorName.toLowerCase();
+		filterState.originalValue = authorName;
+		filterState.displayLabel = `${option.baseLabel}: ${authorName}`;
+	} else if (option.type === "keywords") {
+		const rawKeywords = (await prompt("Enter keywords", "", "text"))?.trim();
```

Comment:
(another, same here as plugins.js) Internationalisation is missing for `Enter keywords`

```suggestion
        const rawKeywords = (await prompt(strings["Enter keywords"] || "Enter keywords", "", "text"))?.trim();
```